### PR TITLE
feat: webui add labels filter for experiments [DET-4117]

### PR DESCRIPTION
**Pull request by apizzini**
## Description

Adds a labels filter to the experiments page.

Default <SelectFilter> placeholder logic behaves a bit tricky in this case because the default case is that no labels are selected, and that causes the Ant Select `maxTagPlaceholder` to don't be rendered (in the other multiple cases, <MetricSelectFilter>, have a min 1 value always selected). For that reason I used the placeholder with an hardcoded "all" value and changed the color to mimic the other placeholder.

Waiting for feedbacks to eventually improve that.

Another improvable point (not sure about the current preferred method) is the TS interface of the experiments labels API, I resolved it to a simple string[].

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

### CODE REVIEW

File: webui/react/src/services/api.ts [hunk 1]
```diff
@@ -113,6 +113,16 @@ export const setExperimentState = async (
   });
 };
 
+export const getAllExperimentLabels = async (): Promise<string[]> => {
+  try {
+    const data = await detApi.Experiments.determinedGetExperimentLabels();
+    return data.labels || [];
```
Comment (hkang1) [inline webui/react/src/services/api.ts:119]: **suggestion:** sort `data.labels` alphabetically with `utils/data.ts` > `alphanumericSorter`

**Comment by hkang1** [comment]
**chore:** Can you add an `All` option up near the top that will essentially act as a "Clear" option? This will show all the entries with no label filters applied.

**Review by hkang1 — COMMENTED** [review]
The picker works great and the code looks good. I have two small requests added to this PR.

1) Sorting the labels alphabetically in the dropdown
2) Add and support and `All` option that clear the dropdown and dismisses it when clicked.

**Comment by apizzini** [comment]
Both items are done :)

**Review by hkang1 — APPROVED** [review]
LGTM!

**nit:** Change the `all` option from lowercase to uppercase `All`? 🙏

**Comment by apizzini** [comment]
Done :)

**Pull request merged**